packages/ember-debug/lib/error.js

```diff
@@ -18,15 +7,12 @@ function ExtendBuiltin(klass) {
   @constructor
   @public
 */
-export default class EmberError extends ExtendBuiltin(Error) {
-  constructor(message) {
-    super();
 
-    if (!(this instanceof EmberError)) {
-      return new EmberError(message);
-    }
+export default EmberError;
+function EmberError(message, code) {
```

Comment:
EmberError unfortunately supports being invoked without using new. This ends up being a less hacky solution. We should deprecate that, so we can eventually switch over entirely though.